File: 01a.py
#! /usr/bin/python3
import sys

DIGITS = set(map(str, range(1, 10)))

with open(sys.argv[1], 'r') as fh:
    calibration = 0
    for line in fh:
        digits = [d for d in line if d in DIGITS]
        add = 10 * int(digits[0]) + int(digits[-1])
        calibration += add
print(calibration)

File: 01a.pl
#!/usr/bin/perl
use warnings;
use strict;
use feature qw{ say };

use ARGV::OrDATA;

my $sum = 0;
while (<>) {
    s/^\D+//;
    s/\D+$//;
    my ($left) = /^(.)/;
    my ($right) = /(.)$/;
    $sum += "$left$right";
}
say $sum;

__DATA__
1abc2
pqr3stu8vwx
a1b2c3d4e5f
treb7uchet
